### PR TITLE
PWN-8092 - high fees warning

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8092

## Description of Work

Check for high fees while swapping and show HighFees warning if fee more than given slippage

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt [hunk 1]
```diff
@@ -102,38 +104,36 @@ class JupiterSwapInteractor(
...
-                // this codes goes to 2.7.0, but needed in 2.7.1 so i removed it temporarily
-//                val isHighFeesNotFound = checkForHighFees(totalFeeAndDeposits, outAmount, slippageValue)
-//                if (isHighFeesNotFound) {
-//                    SwapPriceImpactType.None
-//                } else {
-//                    SwapPriceImpactType.HighFees(currentSlippage)
-//                }
-            }
+        val priceImpact = when {
+            priceImpactPercent.isLessThan(onePercent) -> SwapPriceImpactType.None
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt:126]: {}

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt [hunk 2]
```diff
@@ -165,19 +165,23 @@ class JupiterSwapInteractor(
     }
 
     /**
-     * We need to cast BigIntegers to Double to keep numbers after dot when dividing.
-     * We had an error when we divided 2039280 / 955187 and got only 2 instead of full 2.13
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt:169]: could you save my comments with examples why are we casting BigInteger?

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt [hunk 3]
```diff
@@ -165,19 +165,23 @@ class JupiterSwapInteractor(
...
-        totalFeesAndDeposit: BigInteger,
-        outAmount: BigInteger,
-        slippageValuePercent: Int
-    ): Boolean {
-        val feesValuePercent: Int =
-            (totalFeesAndDeposit.toDouble() / outAmount.toDouble())
-                .times(100)
-                .toInt()
-        return feesValuePercent <= slippageValuePercent
+        keyAppFeeLamportsString: String,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt:175]: you can cast keyAppFeeLamportsString in domain entity when mapping

File: app/src/main/java/org/p2p/wallet/jupiter/repository/model/JupiterSwapRoute.kt [hunk 4]
```diff
@@ -4,6 +4,9 @@ import androidx.annotation.IntRange
 import java.math.BigDecimal
 import java.math.BigInteger
 
+/**
+ * @param keyAppFee fee in lamports of out token
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/jupiter/repository/model/JupiterSwapRoute.kt:8]: I don't think it's very usable to do this way
maybe when you are writing a library it makes sense, but when you find this field and you need to know whats that - comment is near

File: app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt (hunk 3 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt]: changed the type from String to BigInteger and renamed `keyAppFee` to `keyAppFeeInLamports`

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/jupiter/repository/model/JupiterSwapRoute.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/jupiter/repository/model/JupiterSwapRoute.kt]: * Used `// parameter information`
kdoc_no_param_used

* Used `@param in class kdoc`
kdoc_param_used

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**